### PR TITLE
Refactor/avatar

### LINKED ISSUE #1579: Update Avatar component to match Ohana styles
### Link task to story

Story: #1473

### Description

Will need to allow badges other than notification e.g. SEN
Also need to have a larger variant

### PR DESCRIPTION
**Pull request by stevenbister**
### 🧹 Please check off the following before submitting:

- [x] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [ ] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [x] Add milestone

---

## ☑ List all github issues that this PR will close:

- Closes #1579 

## 📝 Description

Refactors and updates the `Avatar` component ready for use within Ohana.

## ⛳️ Current behaviour (updates)

- Only available in one size
- Has inner notification component

## 🚀 New behaviour

- Adds size prop
- Refactors notification prop into `AvatarBadge` child (in the design we're not placing notifications on the avatar)
- Simplifies gradient svg to a span with a gradient background image
- Flattens `scss` and re-works some of the block/modifier naming

## 💣 Is this a breaking change (Yes/No): Yes

I have replaced the `src` prop with `img` as I think it's clearer given this isn't just an image.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible cross-device impacts (iOS, Android)
- [x] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [x] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

> Any additional contextual information, such as relevant links/articles/documentation

### Summary by CodeRabbit

- New Feature: Introduced `AvatarBadge` subcomponent to the `Avatar` module, allowing badges to be added to avatars.
- Refactor: Updated `Avatar` component to accept an `img` prop instead of `src`, improving flexibility and consistency.
- New Feature: Added size options (`m` and `l`) for the `Avatar` component, providing more customization options.
- Refactor: Improved modularity in `Avatar` component by extracting `AvatarInitials`, `AvatarGradient`, and `AvatarBadge` into separate components.
- Test: Added test case for rendering badge text correctly when `AvatarBadge` is used.
- Style: Updated CSS styles for `.c-avatar` component to accommodate new features and changes.

### CODE REVIEW

File: packages/osc-ui/src/components/Avatar/Avatar.spec.tsx [hunk 1]
```diff
@@ -31,31 +31,41 @@ describe('Avatar component', () => {
     });
 
     test('should render fallback initially ', () => {
-        render(<Avatar name={NAME} src={IMAGE_URL} />);
+        render(<Avatar name={NAME} img={IMAGE_URL} />);
         const result = screen.queryByText(initial);
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-ui/src/components/Avatar/Avatar.spec.tsx:35]: Yes... I've done that

File: packages/osc-ui/src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -89,15 +126,53 @@ const AvatarGradient = (props) => {
...
+    (props, forwardedRef) => {
         const { children, className, ...rest } = props;
-        const classes = classNames('c-avatar__badge', className);
+        const { size } = useAvatarContext();
+        const sizeModifier = useModifier('c-avatar__badge', size);
+        const classes = classNames('c-avatar__badge', sizeModifier, className);
 
         return (
-            <Badge className={classes} {...rest} ref={forwardRef}>
+            <Badge className={classes} size={size === 's' ? 's' : 'l'} {...rest} ref={forwardedRef}>
```
[the remark this answers is not in the document]
Comment (jkellerman) [reply packages/osc-ui/src/components/Avatar/Avatar.tsx:136]: Got you.

Just something else, I thought I might mention. When checking the component in Storybook it works fine but whenever I spin up Storybook this error message keeps popping up regarding being unable to access classNames from AutoComplete component, see below.

Screenshot 2023-09-11 at 16 29 27

**Review by jkellerman — COMMENTED** [review]
Just another comment.

File: packages/osc-ui/src/components/Avatar/Avatar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-ui/src/components/Avatar/Avatar.tsx:136]: Yeah I've seen that, pretty sure it got fixed in https://github.com/Open-Study-College/osc/pull/1485 so should go away once rebased etc.

**Review by jkellerman — APPROVED** [review]
LGTM!

**Pull request merged**